includes/collection/class-inbox.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Inbox collection file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Collection;
+
+use Activitypub\Activity\Activity;
+use Activitypub\Activity\Base_Object;
+
+use function Activitypub\object_to_uri;
+use function Activitypub\is_activity_public;
+
+/**
+ * ActivityPub Inbox Collection
+ *
+ * @link https://www.w3.org/TR/activitypub/#inbox
+ */
+class Inbox {
+	const POST_TYPE = 'ap_inbox';
+
+	/**
+	 * Add an activity to the inbox.
+	 *
+	 * @param Activity|\WP_Error $activity The Activity object.
+	 * @param int                $user_id  The id of the local blog-user.
+	 *
+	 * @return false|int|\WP_Error The added item or an error.
+	 */
+	public static function add( $activity, $user_id ) {
+		if ( \is_wp_error( $activity ) ) {
+			return $activity;
+		}
+
+		$item = self::get_by_guid( $activity->get_id() );
+
+		// Check for duplicate activity.
+		if ( ! \is_wp_error( $item ) ) {
+			if ( \is_array( $item ) ) {
+				$item = \current( $item );
+			}
+
+			if ( $item instanceof \WP_Post ) {
+				return $item->ID;
+			}
+		}
+
+		$title      = self::get_object_title( $activity->get_object() );
+		$visibility = is_activity_public( $activity ) ? ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC : ACTIVITYPUB_CONTENT_VISIBILITY_PRIVATE;
+
+		$inbox_item = array(
+			'post_type'    => self::POST_TYPE,
+			'post_title'   => sprintf(
+				/* translators: 1. Activity type, 2. Object Title or Excerpt */
+				\__( '[%1$s] %2$s', 'activitypub' ),
+				$activity->get_type(),
+				\wp_trim_words( $title, 5 )
+			),
+			'post_content' => wp_slash( $activity->to_json() ),
+			// ensure that user ID is not below 0.
+			'post_author'  => \max( $user_id, 0 ),
+			'post_status'  => 'pending',
+			'guid'         => $activity->get_id(),
+			'meta_input'   => array(
+				'_activitypub_object_id'             => object_to_uri( $activity->get_object() ),
+				'_activitypub_activity_type'         => $activity->get_type(),
+				'_activitypub_activity_actor'        => Actors::get_type_by_id( $user_id ),
```

Comment:
Can you tell me more about the role of `post_author` and `_activitypub_activity_actor` for inbox items?

In my testing I had a user and the blog user follow the same account. When it posted the inbox item would only be saved once (as expected) but author and actor type were set to whoever received it first. Should `get_by_guid()` check for the post author as well, so we save one for each follower?

<img width="1141" height="568" alt="Image" src="https://github.com/user-attachments/assets/75f6159e-56b5-480a-adc2-1abd59ba5f8c" />

Reply:
yes, it should save two inbox entries!

Reply:
Maybe I have to rethink the usage of the GUID completely... Shouldn't the GUID be unique?